org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/environments/ExecutionEnvironment.java

```diff
@@ -481,31 +428,38 @@ private Properties getJavaProfileProperties(Bundle bundle, String path) {
 		}
 		// For below Java 9 EE, compiler release option should be disabled by default
 		String property = profile.getProperty(JavaCore.COMPILER_COMPLIANCE);
-		if (property != null) {
-			if (JavaCore.compareJavaVersions(property, JavaCore.VERSION_9) < 0) {
-				profile.setProperty(JavaCore.COMPILER_RELEASE, JavaCore.DISABLED);
-			}
+		if (property != null && JavaCore.compareJavaVersions(property, JavaCore.VERSION_9) < 0) {
+			profile.setProperty(JavaCore.COMPILER_RELEASE, JavaCore.DISABLED);
 		}
 		return profile;
 	}
 
 	private static final String JAVASE = "JavaSE"; //$NON-NLS-1$
+	private static final List<String> PRE_JAVA9_OSGI_EES = List.of( //
+			"OSGi/Minimum-1.0", //$NON-NLS-1$
+			"OSGi/Minimum-1.1", //$NON-NLS-1$
+			"OSGi/Minimum-1.2", //$NON-NLS-1$
+			"JavaSE/compact2-1.8", //$NON-NLS-1$
+			"JavaSE/compact3-1.8", //$NON-NLS-1$
+			"JRE-1.1", //$NON-NLS-1$
+			"J2SE-1.2", //$NON-NLS-1$
+			"J2SE-1.3", //$NON-NLS-1$
+			"J2SE-1.4", //$NON-NLS-1$
+			"J2SE-1.5", //$NON-NLS-1$
+			"JavaSE-1.6", //$NON-NLS-1$
+			"JavaSE-1.7", //$NON-NLS-1$
+			"JavaSE-1.8"); //$NON-NLS-1$
 
 	private String calculateVMExecutionEnvs(Version javaVersion) {
-		StringBuilder result = new StringBuilder("OSGi/Minimum-1.0, OSGi/Minimum-1.1, OSGi/Minimum-1.2, JavaSE/compact1-1.8, JavaSE/compact2-1.8, JavaSE/compact3-1.8, JRE-1.1, J2SE-1.2, J2SE-1.3, J2SE-1.4, J2SE-1.5, JavaSE-1.6, JavaSE-1.7, JavaSE-1.8"); //$NON-NLS-1$
-		Version v = new Version(9, 0, 0);
-		while (v.compareTo(javaVersion) <= 0) {
-			result.append(',').append(' ').append(JAVASE).append('-').append(v.getMajor());
-			if (v.getMinor() > 0) {
-				result.append('.').append(v.getMinor());
-			}
-			if (v.getMajor() == javaVersion.getMajor()) {
-				v = new Version(v.getMajor(), v.getMinor() + 1, 0);
-			} else {
-				v = new Version(v.getMajor() + 1, 0, 0);
-			}
+		List<String> environments = new ArrayList<>(PRE_JAVA9_OSGI_EES);
+		int targetMajor = javaVersion.getMajor();
+		for (int major = 9; major <= targetMajor; major++) {
+			environments.add(JAVASE + '-' + major);
+		}
+		for (int minor = 1; minor <= javaVersion.getMinor(); minor++) {
```

Comment:
I **believe** this loop can be removed, as I haven't seen or heard about JavaSE-10.1, JavaSE-17.2 etc. With that, the code will be much clearer what it is supposed to do.